VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -186,8 +184,17 @@ public static CharSequence trim(CharSequence s) {
         return s.subSequence(start, end);
     }
 
-    public static CharSequence removeDuplicateWhitespace(CharSequence text) {
-        return PATTERN_DUPLICATE_WHITESPACE.matcher(text).replaceAll(" ");
+    public static CharSequence removeDuplicateWhitespaces(CharSequence text) {
+        SpannableStringBuilder builder = new SpannableStringBuilder(text);
+        for (int i = 0; i < builder.length(); i++) {
+            if (builder.charAt(i) == ' ') {
+                int nextIndex = i + 1;
+                if (nextIndex < builder.length() && builder.charAt(nextIndex) == ' ') {
+                    builder.replace(i, nextIndex + 1, " ");
+                }
```

Comment:
How about implementing my suggestion at https://github.com/edx/edx-app-android/pull/681#discussion_r60905737? It would replace all the adjacent whitespace in one go, and the code is also reduced (i.e. no duplicate special case checks for whitespace).